orchestration/src/main/java/com/sap/ai/sdk/orchestration/LLMModuleResultDeserializer.java

```diff
@@ -1,82 +1,45 @@
     super(LLMModuleResult.class);
   }
 
   /**
-   * Deserialize the JSON object into one of the subtypes of the base type.
-   *
-   * <ul>
-   *   <li>If elements of "choices" array contains "delta", deserialize into {@link
-   *       LLMModuleResultStreaming}.
-   *   <li>Otherwise, deserialize into {@link LLMModuleResultSynchronous}.
-   * </ul>
+   * Always deserialize into {@link LLMModuleResultSynchronous} since streaming isn't supported yet.
    *
    * @param parser The JSON parser.
    * @param context The deserialization context.
    * @return The deserialized object.
    * @throws IOException If an I/O error occurs.
    */
+  @Nonnull
   @Override
-  public LLMModuleResult deserialize(JsonParser parser, @Nonnull DeserializationContext context)
+  public LLMModuleResult deserialize(
+      @Nonnull final JsonParser parser, @Nonnull final DeserializationContext context)
       throws IOException {
+    val mapper = (ObjectMapper) parser.getCodec();
+    val rootNode = mapper.readTree(parser);
 
-    // Check if the target type is a concrete class
-    JavaType targetType = context.getContextualType();
-    if (targetType != null && !LLMModuleResult.class.equals(targetType.getRawClass())) {
-      return delegateToDefaultDeserializer(parser, context, targetType);
-    }
-
-    // Custom deserialization logic for LLMModuleResult interface
-    var mapper = (ObjectMapper) parser.getCodec();
-    var rootNode = mapper.readTree(parser);
-    Class<? extends LLMModuleResult> concreteClass = LLMModuleResultSynchronous.class;
-
-    // Inspect the "choices" field
-    var choicesNode = rootNode.get("choices");
-    if (choicesNode != null && choicesNode.isArray()) {
-      var firstChoice = (JsonNode) choicesNode.get(0);
-      if (firstChoice != null && firstChoice.has("delta")) {
-        concreteClass = LLMModuleResultStreaming.class;
-      }
-    }
-
-    // Create a new parser for the root node
-    var rootParser = rootNode.traverse(mapper);
-    rootParser.nextToken(); // Advance to the first token
-
-    // Use the default deserializer for the concrete class
-    return delegateToDefaultDeserializer(rootParser, context, mapper.constructType(concreteClass));
-  }
-
-  /**
-   * Delegate deserialization to the default deserializer for the given concrete type.
-   *
-   * @param parser The JSON parser.
-   * @param context The deserialization context.
-   * @param concreteType The concrete type to deserialize into.
-   * @return The deserialized object.
-   * @throws IOException If an I/O error occurs.
-   */
-  private LLMModuleResult delegateToDefaultDeserializer(
-      JsonParser parser, DeserializationContext context, JavaType concreteType) throws IOException {
-    var defaultDeserializer = context.findRootValueDeserializer(concreteType);
-    return (LLMModuleResult) defaultDeserializer.deserialize(parser, context);
+    return mapper.readValue(rootNode.toString(), LLMModuleResultSynchronous.class);
```

Comment:
As discussed we'll only consider one implementation for the interface during deserialization.
I'm only wondering whether this can be solved in an easier way than dedicated deserializer.